torch/utils/data/dataset.py

```diff
@@ -252,6 +252,41 @@ def __len__(self):
             total += len(d)  # type: ignore
         return total
 
+class ShuffleDataset(IterableDataset[T_co]):
+    r"""Dataset shuffled from the original dataset.
+
+    This class is useful to shuffle the existing dataset. The buffer with buffer_size 
```

Comment:
I think it would be good mentioning `IterableDataset` in here, maybe something like the following:
```suggestion
    This class is useful to shuffle an existing instance of an IterableDataset. The buffer with buffer_size 
```
(lint might fail with this change)